### PR TITLE
Improve modified package detection to work with fixtures folder

### PR DESCRIPTION
**Pull request by KevinFairise2**
### What does this PR do?

For some tests we have non-go files that are used as fixtures in subfolder. This small improved map fixture folder with their parent package, to make sure we run the tests on them if fixtures are changed

### CODE REVIEW

File: tasks/go_test.py [hunk 1]
```diff
@@ -757,6 +758,18 @@ def get_impacted_packages(ctx, build_tags=None):
         for file in files
         if file.endswith(".go") or file.endswith(".mod") or file.endswith(".sum")
     }
+
+    # If non-go modified files are included in a folder contained in a package we consider the
```
Comment (KSerrania) [inline tasks/go_test.py:762]: Seems like this comment is not finished

File: tasks/go_test.py [hunk 2]
```diff
@@ -757,6 +758,18 @@ def get_impacted_packages(ctx, build_tags=None):
...
+    for file in files:
+        if not file.endswith(".go"):
+            formatted_path = Path(os.path.dirname(file)).as_posix()
+            while len(formatted_path) > 0:
+                if glob.glob(f"{formatted_path}/*.go"):
+                    print(f"Found {file} belongs to package {formatted_path}")
```
Comment (KSerrania) [inline tasks/go_test.py:768]:
```suggestion
                    print(f"Found {file} belonging to package {formatted_path}")
```

File: tasks/go_test.py (hunk 1 above)
Comment (KevinFairise2) [reply tasks/go_test.py]: Indeed it was not, I just updated it in https://github.com/DataDog/datadog-agent/pull/22955/commits/43536b66208378573953774be89bca26ff5276f0

**Review by amenasria — APPROVED** [review]
**👏 praise:** ‏Nice feature ! LGTM !

**Pull request merged**